src/gui/entry/EntryView.cpp

```diff
@@ -391,18 +410,22 @@ void EntryView::fitColumnsToContents()
 }
 
 /**
- * Mark icon-only columns as fixed and resize them to their minimum section size.
+ * Mark icon-only columns as fixed and resize them to icon-only section size
  */
 void EntryView::resetFixedColumns()
 {
-    if (!isColumnHidden(EntryModel::Paperclip)) {
-        header()->setSectionResizeMode(EntryModel::Paperclip, QHeaderView::Fixed);
-        header()->resizeSection(EntryModel::Paperclip, header()->minimumSectionSize());
-    }
-
-    if (!isColumnHidden(EntryModel::Totp)) {
-        header()->setSectionResizeMode(EntryModel::Totp, QHeaderView::Fixed);
-        header()->resizeSection(EntryModel::Totp, header()->minimumSectionSize());
+    for (const auto& col : {EntryModel::Paperclip, EntryModel::Totp, EntryModel::PasswordStrength}) {
+        if (!isColumnHidden(col)) {
+            header()->setSectionResizeMode(col, QHeaderView::Fixed);
+
+            // Increase column width, if sorting, to accommodate icon and arrow
+            auto width = ICON_ONLY_SECTION_SIZE;
+            if (header()->sortIndicatorSection() == col
+                && config()->get(Config::GUI_ApplicationTheme).toString() != "classic") {
+                width += 18;
```

Comment:
I am not entirely sure if I find the shape-shifting columns particularly neat or particularly annoying.

Reply:
I like it a lot better than the sort indicator covering the icon. We can let this ride and see how it performs across platforms.